miio/ptxswitch.py

```diff
@@ -0,0 +1,304 @@
+
+        # Returns status Off
+        <-  {PTX Switch}  data= {"result":[0],"id":2474}
+
+        # Turn on all switches
+        ->  {Mi Home App} data= {"id":3381,"method":"SetSwitchAll","params":[1,1,1]}
+
+        # Returns On, On, On
+        <-  {PTX Switch}  data= {"result":[1,1,1],"id":3381}
+
+        # Turn off all switches
+        ->  {Mi Home App} data= {"id":3382,"method":"SetSwitchAll","params":[0,0,0]}
+
+        # Returns off, off, off
+        <-  {PTX Switch}  data= {"result":[0,0,0],"id":3382}
+
+        """
+        # Retrieve properties.
+        properties = AVAILABLE_PROPERTIES[self.model].copy()
+
+        # Querying switch status require [0,0,0] as the request
+        # params. Querying other property require property name
+        # (eg. 'switchname1') as the request param.
+
+        # params for querying status of the switches
+        params_switch_status = list()
+
+        # params for querying other properties
+        params_other = list()
+
+        # Query every switch status
+
+        for k in properties:
+            if k[:5] == "is_on":
+                params_switch_status.append(0)
+            else:
+                params_other.append(k)
+
+        result_1 = self.send(
+            "get_prop",
+            params_switch_status
+        )
+
+        param_count = len(params_switch_status)
+        values_count = len(result_1)
+        if values_count >= param_count:
+            # return values always have one more than requested.
+            # get return values only we want
+            result_1 = result_1[:param_count]
+        else:
+            result_1 = [None, None, None]
+            _LOGGER.debug(
+                "Count (%s) of requested params does not match the "
+                "count (%s) of received values.",
+                param_count, values_count)
+
+        # Query other params
+        # A single request is limited to 1 properties. Therefore the
+        # properties are divided into multiple requests
+
+        result_2 = list()
+        for param in params_other:
+            value = self.send(
+                "get_prop",
+                [param]
+            )
+            if len(value) >= 1:
+                v = value[0]
+                if param[:10] == "switchname":
+                    if isinstance(v, str):
+                        result_2.append(v)
+                    else:
+                        _LOGGER.debug("Unexpected type of switchname")
+                        result_2.append(None)
+
+                else:
+                    result_2.append(v)
+
+            else:
+                result_2.append(None)
+                _LOGGER.debug("Property %s returns None.", param)
+
+        return PtxSwitchStatus(
+            defaultdict(lambda: None,
+                        zip(properties,
+                            result_1 + result_2)))
+
+    def turn_switch(self, index: int, switch_state: int) -> bool:
```

Comment:
Please use bool instead of int for the state, you can always convert between those using `bool(int_value)` and `int(bool_value)`.